idealalphaPackage.R

```diff
@@ -0,0 +1,201 @@
+library(pwr)
+library(ggplot2)
+
+#functions to justify alpha levels.
+#first run all the code. In the beginning are functions incoroporated in later functions
+
+#call idealalpha(n, effsize, prior, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level which gives optimal learning
+#the output will be alpha, power and info, which is the expected learning
+
+#call alphaConfirmation(n, effsize, prior, postdis, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level needed to achieve a given posterior after a significant result
+#the output will be alpha, power, info, and confirm, which is the probability after a positive result
+
+#alphaDisconfirmation(n, effsize, prior, postdis, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#to calculate the alpha level needed to achieve a given posterior after a non-significant result
+#the output will be alpha power and disconfirm, which is the probability after a negative result
+
+#probpower(effsize, prior, postdis, postconf, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4)
+#conduct a probability based power analysis. Here posterior after significance and posterior after non-signficance are fixed
+#the function will then return the sample size and the alpha level needed
+#the output will be alpha, power, disconfirm, confirm, info and n, the sample size per group
+
+
+
+#The next functions are defined to be part of the four main functions
+
+#calculate the probability that the hypothesis is true, after confirming
+#' @export
+postconfirm <- function(power,alpha,prior)
+                          {
+        (power*prior)/(power*prior+alpha*(1-prior))
+}
+
+#calculate the probability that the hypothesis is true, after a nonsignificant finding
+#' @export
+postdisconfirm <- function(power, alpha, prior)
+{
+  (1-power)*prior/((1-alpha)*(1-prior)+(1-power)*prior)
+}
+
+#define power function for t.test, anova and test for correlations based on pwr package
+get_power   <- function(alpha, effsize, n, test = "t-test", type = "two.sample", alternative = "two.sided", k = 4){ #get_power for different alpha levels, two tail independent samples t-test
+if(test == "t-test") {
+  power <- pwr.t.test(n, effsize, alpha, type = type, alternative = alternative)
```

Comment:
if you use functions from different packages, it might be better to use package::function(), e.g. pwr::pwr.t.test(...)

Reply:
Is that a part of style guidelines? Otherwise, I would only use :: if I one package I am using for a function masks the function from another package. Then :: would specifically call the function from the masked package.